packages/grant-explorer/src/features/api/oso.ts

```diff
@@ -4,147 +4,70 @@ import { Hex } from "viem";
 const graphQLClient = new GraphQLClient(osoUrl, {
   headers: {
     authorization: `Bearer ${osoApiKey}`,
   },
 });
-let hasFetched = false;
-
-interface IOSOId {
-  projects_v1: {
-    project_id: Hex;
-  };
-}
+let fetchedProject = "";
 
 export interface IOSOStats {
-  code_metrics_by_project_v1: {
-    contributor_count: number;
-    first_commit_date: number;
+  oso_codeMetricsByProjectV1: {
+    contributorCount: number;
+    firstCommitDate: number;
+    activeDeveloperCount6Months: number;
   };
-  events_monthly_to_project: [
-    {
-      bucket_month: number;
-      amount: number;
-    },
-    {
-      bucket_month: number;
-      amount: number;
-    },
-    {
-      bucket_month: number;
-      amount: number;
-    },
-    {
-      bucket_month: number;
-      amount: number;
-    },
-    {
-      bucket_month: number;
-      amount: number;
-    },
-    {
-      bucket_month: number;
-      amount: number;
-    },
-  ];
 }
 
 export function useOSO(projectGithub?: string) {
   const emptyReturn: IOSOStats = {
-    code_metrics_by_project_v1: {
-      contributor_count: 0,
-      first_commit_date: 0,
+    oso_codeMetricsByProjectV1: {
+      contributorCount: 0,
+      firstCommitDate: 0,
+      activeDeveloperCount6Months: 0,
     },
-    events_monthly_to_project: [
-      {
-        bucket_month: 0,
-        amount: 0,
-      },
-      {
-        bucket_month: 0,
-        amount: 0,
-      },
-      {
-        bucket_month: 0,
-        amount: 0,
-      },
-      {
-        bucket_month: 0,
-        amount: 0,
-      },
-      {
-        bucket_month: 0,
-        amount: 0,
-      },
-      {
-        bucket_month: 0,
-        amount: 0,
-      },
-    ],
   };
   const [stats, setStats] = useState<IOSOStats | null>(null);
 
   const getStatsFor = async (projectRegistryGithub: string) => {
+    fetchedProject = projectRegistryGithub;
     if (osoApiKey === "")
       throw new Error("OpenSourceObserver API key not set.");
-    const queryId = gql`{
-      projects_v1(where: {display_name: {_ilike: "${projectRegistryGithub}"}}
-        distinct_on: project_id
-      ) {
-        project_id
+    const queryVars = {
+      where: {
+        displayName: {
+          _ilike: `${projectRegistryGithub}`,
+        },
+      },
+    };
+    const queryStats = gql`
+      query myQuery($where: Oso_CodeMetricsByProjectV1BoolExp) {
+        oso_codeMetricsByProjectV1(where: $where) {
+          contributorCount
+          firstCommitDate
+          activeDeveloperCount6Months
+        }
       }
-      }`;
+    `;
 
     try {
-      hasFetched = true;
-      const idData: IOSOId = await graphQLClient.request<IOSOId>(queryId);
+      const items: IOSOStats = await graphQLClient.request<IOSOStats>(
+        queryStats,
+        queryVars
+      );
 
-      if (!Array.isArray(idData.projects_v1)) {
-        setStats(emptyReturn);
-        return;
+      if (!Array.isArray(items.oso_codeMetricsByProjectV1)) {
```

Comment:
You could combine this if statement with the else condition on line 70 with something like:
```
if (!items.oso_codeMetricsByProjectV1?.length) {
  throw new Error("no stats returned");
}
```
and then the `const parsedItems` block can be pulled out of that if condition and you don't need to duplicate the error.

That more closely follows the "return early" pattern, which makes things easier to read IMO.

Reply:
cheers, will incorporate!